subsys/pm/Kconfig

```diff
@@ -114,13 +113,15 @@ config PM_DEVICE_RUNTIME
 	  enabled, devices can be suspended or resumed based on the device
 	  usage even while the CPU or system is running.
 
-config PM_DEVICE_RUNTIME_EXCLUSIVE
-	depends on PM_DEVICE_RUNTIME
-	bool "Use only on Runtime Power Management on system suspend / resume"
+config PM_DEVICE_SYSTEM_MANAGED
```

Comment:
Isn't this changing the default behaviour when `PM_DEVICE=y` and `PM_DEVICE_RUNTIME=y`?
Previously hooks wouldn't be enabled, now they are.

Reply:
`PM_DEVICE_RUNTIME_EXCLUSIVE=y` can be replace with `PM_DEVICE_SYSTEM_MANAGED=n`.

I believe this is more clear. 

`PM_DEVICE` is the base and need by RUNTIME and SYSTEM_MANAGED. `PM_DEVICE` is the base to enable pm in a device (aka pm callback)

`PM_DEVICE_RUNTIME` enables device runtime
`PM_DEVICE_SYSTEM_MANAGED` tell the pm subsystem to suspend / resume devices when the soc goes idle.

If one does not want device system managed pm, just disable it. We can make `default n if PM_DEVICE_RIUNTIME`

Reply:
I understand it can be replaced, but `PM_DEVICE_RUNTIME_EXCLUSIVE=y` was  the previous default, and `PM_DEVICE_SYSTEM_MANAGED=y` is the new default, which is the opposite behaviour.

Reply:
Right ... but now we have to explicitly set, in DT, which states triggers system managed device pm.

If we want to keep exactly the same behavior, I can change this pr to;

1) By default, states trigger device pm when `PM_DEVICE_SYSTEM_MANAGED` is set (and users can disable specific sates setting it in DT)
2) PM_DEVICE_SYSTEM_MANAGED is disabled (by default) when PM_DEVICE_RUNTIME is enabled.

With these changes we keep the current behavior. I was already thinking about it (especially due dt changes).
@JordanYates what you think ?